chainntnfs/neutrinonotify/neutrino.go

```diff
@@ -778,6 +779,21 @@ func (n *NeutrinoNotifier) RegisterSpendNtfn(outpoint *wire.OutPoint,
 			neutrino.EndBlock(&headerfs.BlockStamp{
 				Height: int32(ntfn.HistoricalDispatch.EndHeight),
 			}),
+			neutrino.ProgressHandler(func(processedHeight uint32) {
+				// We persist the rescan progress to achieve incremental
+				// behavior across restarts, otherwise long rescans may
+				// start from the beginning with every restart.
+				if processedHeight > lastProcessedHeight {
```

Comment:
is this check needed? Won't this always be increasing?

Reply:
I revisited _UtxoScanner_  and I indeed see that although it handles several scan requests in parallel each scan only moves forward. I will remove this condition.

Reply:
Fixed.